base-math/trunctdsf.c

```diff
@@ -45,7 +45,7 @@ CONVERT_WRAPPER(
 	a_norm = getmantandexpd128 (a, &exp, 15, 1e15DL);
 	/* Handle obvious overflow and underflow to avoid going beyond the
 	   bounds of the exponent table.  */
-	if (exp > 39)		/* Obvious overflow.  */
+	if (exp > __FLT32_MAX_10_EXP__)		/* Obvious overflow.  */
```

Comment:
This macro is only available since GCC 7.  Is that a problem for libdfp?  Should we use 38, instead?

Reply:
Maybe I should create this macro instead

Reply:
If you do, you have to protect the definition with a check for the compiler version, otherwise, your definition will conflict with the definition provided by GCC.  Also, notice that you may not use FLT32_MAX_10_EXP (without the leading and trailing underscores), because that name is reserved by ISO TS-18661.

Reply:
You may also use `__FLT_MAX_10_EXP__` or even `FLT_MAX_10_EXP`, which is provided by `float.h` as required by ISO C11.

Reply:
Ok I'll maybe use `__FLT_MAX_10_EXP__` instead create a macro then.  But, once again is this macro available in GCC < 7?